Makefile

```
@@ -478,6 +478,17 @@ version: $(VERSRC)
 # This rule triggers re-generation of version files specified if Makefile changes.
 $(VERSRC): Makefile
 	VERSION=$(VERSION) $(MAKE) -f version.mk setver
+	$(MAKE) update-api-module-path
+
+# This rule updates the api module path to be in sync with the current api release version.
+# e.g. github.com/gravitational/teleport/api/vX -> github.com/gravitational/teleport/api/vY
+# the go program called will skip update if:
+#  1. The major version hasn't changed - e.g. 7.0.0 -> 7.1.0
+#  2. A suffix is present in the version - e.g. "-alpha"
+.PHONY: update-api-module-path
+update-api-module-path: 
+	@test $(VERSION)
+	./build.assets/api/update-module-path.sh $(VERSION)
```

I don't want to bother you with my preferences but I still think the standalone script can be entirely avoided:

```suggestion
update-api-module-path: MAJ_VER := $(shell echo "$(VERSION)" | cut -d"." -f1)
update-api-module-path: VER_SUFFIX := $(shell echo "$(MAJ_VER)" | awk '$$1 >= 2 {print "\\/v"$$1}')
update-api-module-path:
	test "$(VERSION)" && echo "$(VERSION)" | grep -vq "-"
	find . -type f -iregex '.*\.\(go\|proto\|mod\)' -print0 | xargs -0 sed -i -E "s/(gravitational\/teleport\/api)(\/v[0-9]+)?/\1$(VER_SUFFIX)/"
	sed -i -E "s/(gravitational\/teleport\/api$(VER_SUFFIX)) v[0-9]\+.0.0/\1 v$(MAJ_VER).0.0/" go.mod examples/go-client/go.mod
	$(MAKE) update-vendor grpc
```

---

I agree - for preference I would move all this script logic into the Makefile.

---

I tried to avoid it since the Makefile is already a bit unruly and I didn't want to add to the mountain of scripts. However, since Andrej helped me simplify it (a ton) and it is/has a dependency for other make targets, I think having it in the Makefile ultimately makes more sense (at with how things are currently laid out).